tests/e2e/e2e_setup_test.go

```diff
@@ -391,84 +391,116 @@ func noRestart(config *docker.HostConfig) {
 	}
 }
 
-func (s *IntegrationTestSuite) initUpgrade() {
+func (s *IntegrationTestSuite) upgrade() {
 	// submit, deposit, and vote for upgrade proposal
-	s.submitProposal(s.chains[0])
-	s.submitProposal(s.chains[1])
+	currentHeightA := s.chainHeight(s.valResources[s.chains[0].ChainMeta.Id][0].Container.ID)
```

Comment:
Can we instead move all these calls to a for loop? We can iterate over all chains and do these actions instead of duplicating calls for every chain

Reply:
Can you point out which lines you are referring to?

Reply:
https://github.com/osmosis-labs/osmosis/pull/1458/files/dace939d3b2461b62fbd5c407393b6905339a21d#r869717359